src/app/games/game/game-progress/game-chart/game-chart.component.ts

```diff
@@ -34,14 +34,19 @@ export class GameChartComponent implements OnChanges {
 
   ngOnChanges(): void {
     // don't do anything until all inputs are provided
-    if (typeof this.matching !== "string" || typeof this.nonmatching !== "string" || typeof this.metadata !== "object") {
+    if (/* typeof this.line_names !== "object" || */ typeof this.datasets !== "object" || typeof this.metadata !== "object") {
       return;
     }
 
     // set up chart data
     const metadata = this.metadata;
-    let matchingData = this.parseData(this.matching);
-    let nonmatchingData = this.parseData(this.nonmatching);
+    let nonmatchingData = this.parseData(this.datasets[0]);
+    let matchingData = this.parseData(this.datasets[1]);
+
+    // Names for lines
+    let nonmatchingName = this.metadata.line_names[0];
+    let matchingName = this.metadata.line_names[1];
+
```

Comment:
Could a more appropriate name other than "matching/nonmatching" be used here, as they are not necessarily these anymore?

Reply:
I don't think there is a useful way to do this, espacially since the series are drawn from the matching/nonmatching csvs. Moreover, the primary purpose of this system has always been to show matching/nonmatching; the assets stuff is more of an ancillary effect, which is why the same csvs are used.

Reply:
I'm happy to resolve on the basis of these being the primary names.